Bugfix for bad output, "elapsed time 30 days 24:10 sec"

jquast opened the pull request:
O= on x86-64 Debian/GNU Linux 10 (buster), gcc 8.3.0, with local timezone PDT, output is phrase "30 days 24:10 sec", which is wrong:

```
[16-03-20 20:47:29] Freq: 152.031 MHz active, Level: -28.60/-40.00 b [elapsed time 30 days 24:10 sec]
```

Proposed is a patch that has output of:

```
[17-03-20 23:21:39] Freq: 155.730 MHz active, Level: -45.40/-45.40 b [0h 00m 1.00s elapsed]
```

Thanks and best wishes

jquast committed:
bugfix "elapsed time 30 days 24:10 sec"

I'm not sure if anyone had noticed this before, but at least on my
machine, a x86-64 Debian/GNU Linux 10 (buster), gcc 8.3.0, output
is "30 days 24:10 sec":

[16-03-20 20:47:29] Freq: 152.031 MHz active, Level: -28.60/-40.00 b [elapsed time 30 days 24:10 sec]

This is in error, proposed is a patch that has output of:

[17-03-20 23:21:39] Freq: 155.730 MHz active, Level: -45.40/-45.40 b [elapsed time  0h 0m7.00s]

jquast commented:
I had many more improvements and fixes to submit, but I thought I’d pick the simplest one to see whether the project was active.

I would please suggest if you cannot maintain a project, to make that clearly described at the top of the readme and disable issues/pull requests, it could save the next volunteer a lot of trouble.

lwvmobile commented:
Hey Jeff if you're interested, I've forked this project and put a few little tweaks in it. Its still a bit experimental, but its fixed a few issues with bookmark scanning and also the also the output elapsed time. 

https://github.com/lwvmobile/gqrx-scanner-better-bookmark-scan

jquast commented:
Cool! I’ll be sure to try your fork next time, thank you for Forking :)

I just moved but I hope I can get it all together again soon, like other open issues here discuss, I messed about w/the timing to prevent it from stopping on the wrong channel all the time, added automatic recording, and, implemented a novel Fibonacci scanning technique that improved hitting the same channel on many on/off squelches like would happen in any voice transmission between parties. I’ll try to contribute those the best I can.

I’m now studying gqrx for improvements, there is a significant delay to begin a recording that I’d like to reduce, and, I have an idea for adding a new control code to transfer the waterfall results as a series of ascii floats, so that we don’t have to manually tune, wait, and check squelch — any dummy could see which channels are active by visually inspecting the waterfall, if only we could get a hold of that data I think we could make an excellent scanner, look forward to it!

lwvmobile commented:
Sounds like you are WAY ahead of me, that's for sure. All I've done is a little casual tinkering with the code to keep it from stopping during the bookmark scan all the time and/or landing on the wrong bookmark by slowing the scan speed down and the time issue. Sounds like you've got a lot more worked into your own fixed up version of it.

neural75 commented:
Ok folks, sorry for the delay but as you noticed I'm not actively working on this project because of time and because I lost my SDR dongle during a move to my new apartment so I can't test your fixes (I'm waiting a new one).

So I invited you to participate to the development on the master branch in order to make it simplier for you to proceed with fixes and enhancements.
 
@lwvmobile : I've merged your fixes, seems good to me, also the new cli arguments are fine. Please remove the fork notice from readme appropriately.
@jquast : I had no time to review your fixes but feel free to coordinate with lwvmobile on the conflicting parts and merge it back here where appropriate.

Difftime was a kludge I don't even remember why I wrote it like this so change it as you like :)

I like the lwvmobile approach to extend with specific command line options and so thank you all.
